### ISSUE TITLE
AWS_DEFAULT_PROFILE ignored

### ISSUE THREAD

**Issue by darkn3rd** [aws/aws-sdk-ruby#2034]
Please fill out the sections below to help us address your issue

### Issue description

Ruby AWS-SDK seems to ignore `AWS_DEFAULT_PROFILE` as documented in [Configuring the AWS CLI: Named Profiles](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html).

In looking through the code, noticed this line:

https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/shared_credentials.rb#L31

Which is not consistent to documentation around configuring AWS.  Interim workaround is to do 

```bash
export AWS_PROFILE=$AWS_DEFAULT_PROFILE
ruby buckets.rb
```

Fix 

```ruby
@profile_name ||= ENV['AWS_PROFILE'] || ENV['AWS_DEFAULT_PROFILE']
```

### Gem name 

`aws-sdk-core (3.48.3)`

### Version of Ruby, OS environment

`ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin16]`

### Code snippets / steps to reproduce

```ruby
# buckets.rb
s3 = Aws::S3::Resource.new(region: 'us-west-2')

s3.buckets.limit(50).each do |b|
  puts "#{b.name}"
end
```

```bash
export AWS_DEFAULT_PROFILE=ninjapants
ruby buckets.rb
```
```

**Comment by cjyclaire** [comment]
Thanks for the input! this is already a feature request we are [tracking](https://github.com/aws/aws-sdk-ruby/blob/e36b3efe28644e20f4f296ca11f2dfedc18888a7/FEATURE_REQUESTS.md#add-support-for-cli-profile-environment-variable)

**Comment by mullermp** [comment]
Fixed by #2050
